src/hooks/screen/useChatScreenHook.ts

```diff
@@ -90,10 +93,26 @@ function useChatScreenHook(type: 'SIGNED' | 'ANONYMOUS'): UseChatScreenHook {
       if (type === 'ANONYMOUS') {
         response = await AnonymousMessageRepo.sendAnonymousMessage(selectedChannel?.id, message);
       } else {
+        const allAttachmentPromises: Promise<void>[] = attachments.map(async (item) => {
+          if (item.type === 'image') {
+            const uploadedImageUrl = await ImageUtils.uploadImageWithoutAuth(item.asset_url);
+            return {
+              ...item,
+              asset_url: uploadedImageUrl.data.url,
+              thumb_url: uploadedImageUrl.data.url
+            };
+          }
+
+          return item;
+        });
+
+        const newAttachments = await Promise.all(allAttachmentPromises);
+
         response = await SignedMessageRepo.sendSignedMessage(
           selectedChannel?.id,
           message,
-          channelType
+          channelType,
+          newAttachments
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

This block of code uploads images without authentication. This could potentially lead to security issues if unauthorized users gain access to upload images. Consider adding authentication for image uploads.

Also, the same URL is used for both `asset_url` and `thumb_url`. If the thumbnail is supposed to be a smaller version of the image, you might need to generate a separate thumbnail image and upload it separately.

Lastly, there's no error handling for failed image uploads. Consider adding a try-catch block around the image upload and handle any errors that may occur.

```diff
-         const allAttachmentPromises: Promise<void>[] = attachments.map(async (item) => {
-           if (item.type === 'image') {
-             const uploadedImageUrl = await ImageUtils.uploadImageWithoutAuth(item.asset_url);
-             return {
-               ...item,
-               asset_url: uploadedImageUrl.data.url,
-               thumb_url: uploadedImageUrl.data.url
-             };
-           }
- 
-           return item;
-         });
- 
-         const newAttachments = await Promise.all(allAttachmentPromises);
+         const allAttachmentPromises: Promise<void>[] = attachments.map(async (item) => {
+           try {
+             if (item.type === 'image') {
+               const uploadedImageUrl = await ImageUtils.uploadImageWithAuth(item.asset_url); // use authenticated upload
+               const thumbnailUrl = await ImageUtils.generateThumbnail(item.asset_url); // generate thumbnail
+               return {
+                 ...item,
+                 asset_url: uploadedImageUrl.data.url,
+                 thumb_url: thumbnailUrl.data.url
+               };
+             }
+           } catch (error) {
+             console.error('Failed to upload image:', error);
+           }
+ 
+           return item;
+         });
+ 
+         const newAttachments = await Promise.all(allAttachmentPromises);
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->